src/components/File/hooks/use-file-by-filepath.ts

```diff
@@ -52,6 +52,7 @@ export const useFileByFilepath = () => {
   });
 
   const [debouncedEditor] = useDebounce(editor?.state.doc.content, 4000);
+  const [lastSavedContent, setLastSavedContent] = useState<string>("");
```

Comment:
Content checking is an o(n) operation, which isn't a big deal now imo but could be optimized fairly easily. Instead of checking the content directly, could we configure a simple listener directly on the editor that checks if the user did update the text recently? 

The state can be managed in a boolean called `isFileContentModified` which is fired on change to the file content, and it could be changed back to false whenever the file has been saved and indexed. This way we can hold very little content in memory.

Reply:
Good idea! This is now implemented with ```isFileContentModified```